tests/contracts/nonfungible-token/contract.scilla

```diff
@@ -0,0 +1,411 @@
+(***************************************************)
+(*               Associated library                *)
+(***************************************************)
+import BoolUtils
+
+library NonfungibleToken
+
+let one_msg = 
+  fun (msg : Message) => 
+  let nil_msg = Nil {Message} in
+Cons {Message} msg nil_msg
+
+(* Checks and see if an address is a contract owner *)
+let checkContractOwner =
+    fun (msgSender: Address) => 
+    fun (contractOwner: Address) =>
+        let isOwner = builtin eq msgSender contractOwner in
+        match isOwner with
+        | True => True
+        | False => False
+        end
+
+(* Checks and see if an address is a token owner *)
+let isTokenOwner = 
+    fun (msgSender: Address) =>
+    fun (tokenId: Uint256) =>
+    fun (tokenOwnerMap_tmp : Map (Uint256) (Address)) =>
+        let tokenOwner = builtin get tokenOwnerMap_tmp tokenId in
+        match tokenOwner with
+        | None => False
+        | Some val =>
+            let isSenderTokenOwner = builtin eq val msgSender in 
+            match isSenderTokenOwner with
```

Comment:
Similar to above, `match` can be replaced with just it's argument (`builtin eq val msgSender`).